engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/AbstractTableLocation.java

```diff
@@ -87,24 +87,31 @@ public LivenessReferent asLivenessReferent() {
     // TableLocationState implementation
     // ------------------------------------------------------------------------------------------------------------------
 
+    protected void initializeState() {
+        // No-op by default, can be overridden by subclasses to initialize state on first access
+    }
```

Comment:
It probably makes sense to add javadoc that the expectation for static locations that override this is to call `handleUpdateInternal`.